Release notes review (#204)
iomesh/docs: docs/appendices/release-notes.md

sidebar_label: Release Notes
---

## IOMesh 1.0.0 Release Notes

### What's New

#### New Features
**Installation & Deployment**

> - Adds support for deploying IOMesh on the Hygon x86_64 and Kunpeng AArch64 architectures.

Reviewer: ```suggestion
- Supports for deploying IOMesh on the Hygon x86_64 and Kunpeng AArch64 architectures.
```

> - Adds support for Kubernetes version 1.25.

Reviewer: ```suggestion
- Supports for Kubernetes version 1.25.
```

- Enables deployment of multiple IOMesh clusters within a single Kubernetes cluster.

**Storage**

> - Allows for creating PVs using local storage like a directory or block device for pod use.

Reviewer: ```suggestion
- Allows for dynamically  provisioning PVs using local storage like a directory or block device for pod use.
```

- Implements CSI typology feature to ensure that when there are multiple IOMesh clusters, pods that use PVs from a specific cluster can be scheduled on the worker node that holds that cluster.
- Triggers pod high availability when its worker node has a power outrage.

**Operations & Management**

> - Detects abnormal disks and isolates them to minimize the impact on system performance and reduce operational burden.

Reviewer 2: ```suggestion
- Detects abnormal disks and isolates unhealthy disks to minimize the impact on system performance and reduce operational burden.
```

Author: 是不是写成 disks in `Unhealthy` state 好一些？因为字面上看，abnormal 和 unhealthy 区别不大，我们想说的是 abnormal 有一类状态是 `Unhealthy` ，处于这种状态才会被隔离

Reviewer 2: 可以的

#### Improved Features

**Storage**
- Reduced IO interruption to within 22 seconds in case of node or network failure, ensuring that data access is timely and reliable even in the face of disruptions.
- Enhanced data integrity by assigning a temporary replica to hold newly written data after a replica has been removed. 
- Enhanced data channel fault tolerance to prevent disconnection due to IO timeout.
- Optimized Lease Owner allocation mechanism to avoid IO failures due to network failures, enabling more efficient and robust data transmission.

> - Cleared the file lock created when executing `iscsiadm`, or else the CSI driver will be unavailable.

Reviewer: ```suggestion
- Cleared the file lock created when executing `iscsiadm`, or else the CSI Driver will be unavailable.
```

Author: 这个只有叫 IOMesh CSI Driver, Driver 才大写？

- Simplified the configuration method for accessing IOMesh from outside its Kubernetes cluster, eliminating the need to create a separate Kubernetes service and making it easier to access and use IOMesh.
- Optimized the default CPU/memory resource limit setting for pods to avoid pods running slowly due to insufficient resources.

**Operations & Management**

- Added an alert panel on the IOMesh dashboard and display of information about cluster, physical disk, and PV, providing administrators with real-time monitoring and status updates.

#### Resolved Issues

**Storage**

- Meta DB space become full after a snapshot of a large volume was taken, making the ZooKeeper unavailable. This issue has been resolved in this version.
- After the chunk IP was changed, the data channel manager was unable to detect the new chunk IP, resulting in data migration failure. The issue has been resolved in this version.

**Operations & Management**
- Uninstalling IOMesh via `Helm` sometimes left pod resources behind, which has been resolved in this version.
  
### Specifications

| Component | Version|
| -------| -------|
|iomesh-operator|1.0.0|

> | csi-driver| 2.6.0|

Reviewer 2: ```suggestion
|csi-driver| 2.6.0|
```

|zbs|5.3.0|
|zookeeper|3.5.9|
|node-disk-manager|1.8.0|
|hostpath-provisioner|0.5.3|

### Compatibility

#### Server Architecture Compatibility for IOMesh  

IOMesh is compatible with Intel x86_64 and Hygon x86_64 or Kunpeng AArch64 architectures.

#### Kubernetes and Linux OS Compatibility for IOMesh

<table>
<thead>
<tr class="header">

> <th>Item</th>

Author: done

<th>Version</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Kubernetes</td>
<td>
<ul>
<li>Kubernetes v1.17-v1.25</li>
<li>OpenShift v3.11-v4.10</li>
</ul></td>
</tr>
<tr class="even">
<td>Linux OS</td>
<td>
<ul>
<li>CentOS 7</li>
<li>CentOS 8</li>
<li>CoreOS</li>
<li>RHEL 7</li>
<li>RHEL 8</li>
<li>OpenEluer 22.03 LTS</li>
</tbody>
</table> 

>**Note**:
>
> IOMesh has no dependencies on the Linux OS version. The versions listed above are tested versions only.

### Known Issues

- If a Kubernetes cluster has multiple IOMesh clusters deployed, the IOMesh dashboard will display alert information of all clusters instead of just one.
- The IOMesh dashboard cannot display the correct storage capacity usage when the IOMesh cluster has invalid storage capacity.
- A PV with `Block` as the volume mode can still be written due to a Kubernetes code defect even if the access mode is ReadOnlyMany.

> - The following issues persist due to a flaw in the NDM mechanism:

Reviewer: ```suggestion
- The following issues persist due to a flaw in the NDM(Node Disk Manager) mechanism:
```

Author: 这里需要给一个 NDM的链接吗

>   - After an IOMesh cluster is unmounted, the disk corresponding to the block device used by this cluster might not be cleared completely.

Reviewer 2: ```suggestion
  - After an IOMesh cluster is uninstalled, the disk corresponding to the block device used by this cluster might not be cleared completely.
```

>   - After an IOMesh cluster is unmounted, the disk corresponding to the block device used by this cluster might not be cleared completely.

Reviewer: ```suggestion
  - After an IOMesh cluster is uninstalled, the disk corresponding to the block device used by this cluster might not be cleared completely.
```

>   - After the IOMesh Device Local PV is deleted, the disk corresponding to the block device used by this cluster might not be cleared completely.

Reviewer: 这里要不要和上面 的 pv 一样用 an IOMesh Local PV with `device`呀

Author: 咱们在 Local PV Manager 直接说的 IOMesh device local PV 和 IOMesh hostpath local PV, 应该可以这么说？

>   - After the IOMesh Device Local PV is deleted, the disk corresponding to the block device used by this cluster might not be cleared completely.

Reviewer: ```suggestion
  - After the IOMesh Device Local PV is deleted, the disk corresponding to the block device might not be cleared completely.
```

  - The state of a block device might switch back and forth between `active` and `inactive`, which sometimes causes the disk to be unmountable.
- After removing the chunk pod on a worker node, IOMesh CR still shows disks on this node.

> - After an IOMesh cluster is uninstalled, the metrics in the IOMesh Operator may not be completely cleaned up. This can cause the license version to not display in the dashboard if you later deploy an IOMesh cluster with the same name in the same namespace. You may restart the Operator pod to fix this issue.

Reviewer 3: This can cause the license version to not display in the dashboard 中 时license edition not display
